src/d3a/models/strategy/storage.py

```diff
@@ -291,8 +284,7 @@ def event_market_cycle(self):
         if ConstSettings.IAASettings.MARKET_TYPE == 2 or \
            ConstSettings.IAASettings.MARKET_TYPE == 3:
             self.state.clamp_energy_to_buy_kWh([current_market.time_slot])
-            self.update_market_cycle_bids(final_rate=self.break_even[
-                self.area.now].buy)
+            self.bid_update.update_market_cycle_bids(self)
```

Comment:
You are calling this line twice in this function (also in line 272). But really, you only want to call it for the two sided markets, so ONLY here, right?